### PR TITLE
Add support for modified buffers

### LINKED ISSUE #1: Support modified files
It would be nice if this tool could accept modified files on stdin instead of requiring editors to save them off as temp files.

I would recommend using the same _archive_ format as `guru` and `gogetdoc`, which has a parser in  the [`go/buildutil` package](https://godoc.org/golang.org/x/tools/go/buildutil#ParseOverlayArchive).

The changes should be pretty straightforward:
- document the archive format in the readme
- add a `-modified` flag, and if present, use `buildutil.ParseOverlayArchive` to read the archive from stdin
- if the file is present in the archive, populate the `src` argument of [`parser.ParseFile`](https://godoc.org/go/parser#ParseFile)

With these updates, you'll be able to remove the temp files from vim-go and reuse the same code you already use for `guru` and `gogetdoc`.

I'm happy to take a stab at implementing these changes if you agree with the idea.

### PR DESCRIPTION
**Pull request by zmb3**
Modified files can be read from stdin using the same archive format as the guru and gogetdoc tools.

Fixes #1

### CODE REVIEW

**Comment by zmb3** [comment]
I'm thinking since gomodifytags only ever operates on a single file, we can ditch the archive format and just write the contents of the file to stdin, which simplifies things even further. 

What do you think @fatih?

File: main.go [hunk 1]
```diff
@@ -168,7 +176,17 @@ func realMain() error {
...
 	c.fset = token.NewFileSet()
-	node, err := parser.ParseFile(c.fset, c.file, nil, parser.ParseComments)
+	var contents interface{}
+	if c.modified != nil {
+		archive, err := buildutil.ParseOverlayArchive(c.modified)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse -modified archive: %v", err)
+		}
+		if fc, ok := archive[c.file]; ok {
+			contents = fc
```
Comment (fatih) [inline main.go]: What if `c.file` doesn't exist here? Can you also return error in that case and continue

**Comment by fatih** [comment]
> I'm thinking since gomodifytags only ever operates on a single file, we can ditch the archive format and just write the contents of the file to stdin, which simplifies things even further.

It's a good idea to simplify it, however we're going to end up with a custom archive format. I think the current solution is good and I like it that way.

File: main.go (hunk 1 above)
Comment (zmb3) [inline main.go]: Good catch! Should be all set now.

**Comment by fatih** [comment]
Thanks @zmb3

**Pull request merged**